relay-server/src/metrics_extraction/generic.rs

```diff
@@ -23,22 +23,28 @@ pub trait Extractable: Getter {
 /// The instance must have a valid timestamp; if the timestamp is missing or invalid, no metrics are
 /// extracted. Timestamp and clock drift correction should occur before metrics extraction to ensure
 /// valid timestamps.
-pub fn extract_metrics<T>(instance: &T, config: &MetricExtractionConfig) -> Vec<Bucket>
+pub fn extract_metrics<T>(instance: &T, config: CombinedMetricExtractionConfig) -> Vec<Bucket>
 where
     T: Extractable,
 {
     let mut metrics = Vec::new();
+    let mut mris_seen = BTreeSet::new();
 
     let Some(timestamp) = instance.timestamp() else {
         relay_log::error!("invalid event timestamp for metric extraction");
         return metrics;
     };
 
-    for metric_spec in &config.metrics {
+    for metric_spec in config.metrics() {
         if metric_spec.category != instance.category() {
             continue;
         }
 
+        let is_new = mris_seen.insert(metric_spec.mri.as_str());
+        if !is_new {
+            continue;
+        }
```

Comment:
I added this logic because with multiple sources of MRI definitions, I expect accidental duplicates to become more probable. @jan-auer let me know if this goes against the original design.

Reply:
We could move this check to right before `metrics.push(Bucket` to allow multiple non-overlapping definitions (first match wins, like with tags).

Reply:
Solved, we need to support multiple occurrences (e.g. on-demand metrics).